app/src/main/java/org/kiwix/kiwixmobile/intro/CustomViewPager.kt

```diff
@@ -0,0 +1,62 @@
+/*
+ * Kiwix Android
+ * Copyright (c) 2019 Kiwix <android.kiwix.org>
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+package org.kiwix.kiwixmobile.intro
+
+import android.content.Context
+import android.util.AttributeSet
+import android.view.animation.Interpolator
+import android.widget.Scroller
+import androidx.viewpager.widget.ViewPager
+
+/**
+ * A custom implementation of [ViewPager] to decrease the speed of auto-scroll animation
+ * of [ViewPager].
+ */
+class CustomViewPager : ViewPager {
+  constructor(context: Context?) : super(context!!) {
+    postInitViewPager()
+  }
+
+  constructor(context: Context?, attrs: AttributeSet?) : super(context!!, attrs) {
+    postInitViewPager()
+  }
+
+  companion object {
+    private const val CUSTOM_DURATION = 3
+  }
+
+  /**
+   * Override the [Scroller] instance with our own class so we can change the
+   * duration
+   */
+  private fun postInitViewPager() {
+    val scroller = ViewPager::class.java.getDeclaredField("mScroller")
```

Comment:
This can still very easily throw an exception, the `try catch` block should have been kept.

Kotlin's lack of checked exceptions mean these can really sneak up on you.

Reply:
@macgills Getting an error TooGenericExceptionCaught while gradle build, any options?

Reply:
@macgills Is it ok if use { @SuppressLint("TooGenericExceptionCaught") } to the function?

Reply:
I would prefer the specific exceptions be caught, catching Exception is generally bad practice

Reply:
Done👍, please have a look.